stan/math/rev/core/precomputed_gradients.hpp

```diff
@@ -165,7 +166,7 @@ class precomputed_gradients_vari_template : public vari {
   template <typename Op, typename Grad,
             require_all_not_std_vector_t<Op, Grad>* = nullptr>
   void chain_one(Op& op, const Grad& grad) {
-    op.vi_->adj_ += this->adj_ * grad;
+    op.vi_->adj_ = op.vi_->adj_ + this->adj_ * grad;
```

Comment:
Is this because we don't have a += operator for matrix_cl? Should we?

Reply:
Yes and probably. When adding that we should probably also add it to kg expressions that can function as lvalues.

Reply:
Yeah let's do that so that we can keep using the `+=` stuff

Reply:
Lets make it a separate PR.